pkg/controller/replicaset/replica_set.go

```diff
@@ -802,10 +804,28 @@ func getPodsToDelete(filteredPods, relatedPods []*v1.Pod, diff int) []*v1.Pod {
 	if diff < len(filteredPods) {
 		podsWithRanks := getPodsRankedByRelatedPodsOnSameNode(filteredPods, relatedPods)
 		sort.Sort(podsWithRanks)
+		filteredPods = podsWithRanks.Pods
+		reportMetrics(filteredPods, diff)
 	}
 	return filteredPods[:diff]
 }
 
+func reportMetrics(filteredPods []*v1.Pod, diff int) {
+	youngestTime := time.Time{}
+	// first we need to check all of the pods to get the youngest, as they may not necessarily be sorted by timestamp alone
+	for _, pod := range filteredPods {
+		if time.Since(pod.CreationTimestamp.Time) < time.Since(youngestTime) {
+			youngestTime = pod.CreationTimestamp.Time
+		}
+	}
+
+	// for each pod chosen for deletion, report the ratio of its age to the youngest pod's age
+	for _, pod := range filteredPods[:diff] {
+		ratio := time.Since(pod.CreationTimestamp.Time).Seconds() / time.Since(youngestTime).Seconds()
```

Comment:
this could very well be a zero denominator, as `pod.CreationTimestamp` has granularity of seconds :(

Reply:
Switching this to a single `time.Now` as mentioned above and using milliseconds should fix that right?